tools/ReleaseTools.psm1

```diff
@@ -332,28 +345,25 @@ function New-DraftRelease {
         [ValidateSet([RepoNames])]
         [string]$RepositoryName
     )
+    param(
+        [Parameter(ValueFromPipeline)]
+        [string[]]$Assets
+    )
     $Version = Get-Version -RepositoryName $RepositoryName
     $Changelog = (Get-FirstChangelog -RepositoryName $RepositoryName) -join "`n"
     $ReleaseParams = @{
         Draft      = $true
+        # NOTE: We rely on GitHub to create the tag.
         Tag        = "v$Version"
+        Committish = "release/v$Version"
         Name       = "v$Version"
         Body       = $ChangeLog
         PreRelease = [bool]$Version.PreReleaseLabel
         # TODO: Pass -WhatIf and -Confirm parameters correctly.
     }
 
-    if ($PSCmdlet.ShouldProcess("$RepositoryName/v$Version", "git tag")) {
-        # NOTE: This a side effect neccesary for Git operations to work.
-        Push-Location -Path "$PSScriptRoot/../../$RepositoryName"
-        if (-not (git show-ref --tags "v$Version")) {
-            git tag "v$Version"
-        } else {
-            Write-Warning "git tag $RepositoryName/v$Version already exists!"
-        }
-        Pop-Location
-    }
+    $Repo = Get-GitHubRepository -OwnerName PowerShell -RepositoryName $RepositoryName
+    $Release = $Repo | New-GitHubRelease @ReleaseParams
```

Comment:
Piping to `New-GitHubRelease` will only take the URI from the input I think ([source](https://github.com/microsoft/PowerShellForGitHub/blob/0ebf9e8b8d0f4fa361996674dbca804da36bace7/GitHubReleases.ps1#L218-L378)).

It's easier to just to add OwnerName and RepositoryName to `$ReleaseParams` than to invoke `Get-GitHubRepository` I'd say.

Reply:
Fair.